### PR TITLE
#3512 - Review Impact of All Restrictions

### PR DESCRIPTION
**Pull request by lewischen-aot**
- Created a DB migration to update the `sims.restriction_action_types` enum to include "Stop part time BC funding".
  - Used "part time" without capitalized letters for "Stop part time BC funding" to be consistent with existing convention across FT/PT restrictions.
- Created a DB migration to update the `sims.restrictions`. Updated restrictions per the spreadsheet to ensure all behaviors and names are updated. ALL CHANGES ARE IN HIGHLIGHTED CELLS
  - Updated `action_type` for the restriction code _5_, _7_ and _9_.
  - Updated `action_type` for the restriction code _12_, _O_ and _PTWTHD_.
  - Updated `restriction_code`, `action_type` for the restriction code _APPC_.
  - Updated `restriction_type`, `description`, `restriction_category` for the restriction code _B6_.
  - Updated `description`, `action_type` for the restriction code _B6A_.
  - Updated `restriction_type`, `restriction_category` for the restriction code _E2_.
  - Updated `action_type` for the restriction code _PTSSR_.
  - Updated `action_type` for the restriction code _SSR_.
  - Updated `action_type` for the restriction code _RD_.
- NOTE There are other updates to when the _PTWTHD_ restriction is applied and accordion behavior for _PTSSR_. (Effects are updated within this spreadsheet)
- Note ID #8 was previously _APPC_ and the name is being changed to _HOLD_ (**line 9 on the spreadsheet**)

#### Rollback evidence for migration AddStopPartTimeBCFundingRestrictionActionType1726275056327
image

#### Rollback evidence for migration UpdateRestrictionsReviewImpact1726275219662
image

### CODE REVIEW

File: sources/packages/backend/apps/db-migrations/src/sql/Restrictions/Rollback-update-restrictions-review-impact.sql [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    description = 'Bankruptcy restriction on file prevent CSL/BCSL disbursements.',
+    restriction_category = 'Federal'
+WHERE
+    restriction_code = 'B6';
+
+-- Revert the updates for the restriction code B6A
+UPDATE
+    sims.restrictions
+SET
+    description = 'For reinstatment of your BCSL eligibility please complete an appeal.',
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/db-migrations/src/sql/Restrictions/Rollback-update-restrictions-review-impact.sql:40]: Please fix the typo "reinstatement". No business confirmation is needed.

**Review by andrewsignori-aot — APPROVED** [review]
Great work, looks good 👍

File: sources/packages/backend/apps/db-migrations/src/sql/Restrictions/Rollback-update-restrictions-review-impact.sql (hunk 1 above)
Comment (lewischen-aot) [reply sources/packages/backend/apps/db-migrations/src/sql/Restrictions/Rollback-update-restrictions-review-impact.sql]: Updated.

File: sources/packages/backend/apps/db-migrations/src/sql/Types/Rollback-add-stop-part-time-bc-funding-restriction-action-type.sql [hunk 2]
```diff
@@ -0,0 +1,30 @@
+--Removing the value Application from sims.note_types for the rollback.
```
Comment (sh16011993) [inline sources/packages/backend/apps/db-migrations/src/sql/Types/Rollback-add-stop-part-time-bc-funding-restriction-action-type.sql:1]: I believe this comment needs to be updated.

Comment (lewischen-aot) [reply sources/packages/backend/apps/db-migrations/src/sql/Types/Rollback-add-stop-part-time-bc-funding-restriction-action-type.sql:1]: Nice catch!

**Review by sh16011993 — APPROVED** [review]
Nice work @lewischen-aot 👍 

Just one comment related comment.

**Pull request merged**